typings/config.d.ts

```diff
@@ -7,29 +7,30 @@
 import Gatherer = require('../lighthouse-core/gather/gatherers/gatherer.js');
 import Audit = require('../lighthouse-core/audits/audit.js');
 
-
 declare global {
   module LH {
     /**
      * The full, normalized Lighthouse Config.
      */
-    export interface Config {
+    export interface Config extends Config.Json {
       settings: Config.Settings;
-      passes?: Config.Pass[];
-      audits?: Config.AuditDefn[];
-      categories?: Record<string, Config.Category>;
-      groups?: Record<string, Config.Group>;
+      passes: Config.Pass[] | null;
+      audits: Config.AuditDefn[] | null;
+      categories: Record<string, Config.Category> | null;
+      groups: Record<string, Config.Group> | null;
```

Comment:
we had a mixture of optional and `null` used when one of these wasn't provided (e.g. don't need `audits` when in `gatherMode`). Went with the null as otherwise we'd need to be deleting things when e.g. everything ends up filtered out or whatever